src/components/Breadcrumbs/CurrentBreadcrumbItem.tsx

```diff
@@ -32,71 +22,42 @@ export const CurrentBreadcrumbItem = ({
         },
         ref,
     );
 
-    const classNames = merge([decorator && 'tw-flex tw-gap-x-1 tw-items-start tw-py-1', bold && 'tw-font-bold']);
+    const elementTypeProps = { a: { href: link }, button: { onClick, type: 'button' as const }, span: {} };
+
     const { isFocusVisible, focusProps } = useFocusRing();
-    const props = mergeProps(itemProps, focusProps);
+    const props = mergeProps(itemProps, focusProps, elementTypeProps[Element]);
+
+    const classNames = merge([
+        'tw-flex tw-gap-x-1 tw-items-center tw-leading-4 tw-h-6',
+        bold && 'tw-font-bold',
+        isFocusVisible && FOCUS_STYLE,
+    ]);
 
     return (
         <li
-            className="tw-w-full tw-inline-flex tw-align-middle tw-mt-1 tw-gap-x-1 tw-text-m tw-text-black dark:tw-text-white"
+            className="tw-w-full tw-inline-flex tw-align-middle tw-gap-x-1 tw-text-m tw-text-text tw-items-center"
             data-test-id="breadcrumb-item"
         >
-            {contentElementType === 'a' && (
-                <ItemWithBadges badges={badges}>
-                    <a
-                        ref={ref as RefObject<HTMLAnchorElement>}
-                        {...props}
-                        href={link}
-                        className={merge([classNames, isFocusVisible ? FOCUS_STYLE : ''])}
-                    >
-                        {decorator}
-                        <span className={merge(['tw-leading-4', bold ? 'tw-font-bold' : ''])}>{label}</span>
-                    </a>
-                </ItemWithBadges>
-            )}
-            {contentElementType === 'button' && (
-                <ItemWithBadges badges={badges}>
-                    <button
-                        ref={ref as RefObject<HTMLButtonElement>}
-                        {...props}
-                        type="button"
-                        onClick={onClick}
-                        className={merge(['tw-leading-4', classNames, isFocusVisible ? FOCUS_STYLE : ''])}
-                    >
-                        {decorator}
-                        {label}
-                    </button>
-                </ItemWithBadges>
-            )}
-            {contentElementType === 'span' && (
-                <>
-                    {decorator}
-                    <ItemWithBadges badges={badges}>
-                        <span
-                            ref={ref as RefObject<HTMLSpanElement>}
-                            {...props}
-                            className={merge([
-                                'tw-leading-4 tw-py-1 ',
-                                bold && 'tw-font-bold',
-                                isFocusVisible && FOCUS_STYLE,
-                            ])}
-                        >
-                            {label}
-                        </span>
-                    </ItemWithBadges>
-                </>
-            )}
+            <Element ref={ref} {...props} href={link} className={classNames}>
```

Comment:
Shouldn't we remove the `href` prop here since it is already part of the `{...props}`? Otherwise we're gonna have `<span href="...">` and `<button href="...">`

Reply:
Good catch! Missed this prop